forte/data/readers/classification_reader.py

```diff
@@ -0,0 +1,239 @@
+
+        # it determines the order of concatenation
+        subtext_fields = self.configs.subtext_fields
+
+        # get text and subtext indices
+        text, subtext_indices = generate_text_n_subtext_indices(
+            subtext_fields, df_dict
+        )
+        self.set_text(pack, text)
+        if df_dict["label"].isdigit() != self.configs.digit_label:
+            raise ProcessorConfigError(
+                "Label format from dataset "
+                "is not consistent with the label format from configs. \n"
+                + "dataset digit label status: "
+                + str(df_dict["label"].isdigit())
+                + "\n"
+                "config digit label status: " + str(self.configs.digit_label)
+            )
+
+        if self.configs.digit_label:
+            if self.configs.one_based_index_label:
+                # adjust label encoding if dataset digit label is one-based
+                class_id = int(df_dict["label"]) - 1
+            else:
+                class_id = int(df_dict["label"])
+        else:
+            # decode string label from dataset into digit label
+            class_id = self._class2index[df_dict["label"]]
+        # initialize all subtexts instances as data pack module
+        # to add corresponding variables and functions
+        for subtext_field, (start_idx, end_idx) in subtext_indices.items():
+            path_str, module_str = subtext_field.rsplit(".", 1)
+            mod = importlib.import_module(path_str)  # sentence ontology module
+            entry_class = getattr(mod, module_str)
+            entry_class(pack, start_idx, end_idx)
+        # for now, we use document to store concatenated text and set the class here
+        doc = Document(pack, 0, subtext_indices[subtext_fields[-1]][1])
+        doc.document_class = [self.configs.index2class[class_id]]
+
+        pack.pack_name = line_id
+        yield pack
+
+    @classmethod
+    def default_configs(cls):
+        r""" "This defines a basic configuration structure for classification dataset reader.
+
+        Here:
+            - data_fields: these fields provides one-to-one correspondence
+                between given dataset fields and ontology paths.
+            - index2class: mapping from indices to classes
+            - subtext_fields: ordered subtext fields that the user want to
+                concatenate into an input text in the same order
+            - digit_label:  specify whether label in dataset is digit
+            - text_label: specify whether label in dataset is text
+            - one_based_index_label: if dataset provides digit label,
+                check if it's one-based index, if so set it to True.
+            - skip_first_line: many datasets' first line are columns names,
+                set this config to True if it's the case.
+        """
+        config: dict = super().default_configs()
+        config.update(
```

Comment:
we don't do `config.update` anymore, should directly return the new set of configs.